src/Basis/Constructors.jl

```diff
@@ -47,7 +171,7 @@ trueweights = [
 
 ```
 """
-function gaussquadrature(q::Int)
+function gaussquadrature(q::Int; mapping = nothing)
```

Comment:
Is there a type we can attach here? Maybe `mapping::function`? Just to help us use it right in the future

Reply:
@jeremylt what do you mean we do here ```mapping::function```?

Reply:
`q` has the type `Int`. The argument `mapping` should also have a type.

Reply:
@jedbrown upon setting ```mapping::function``` in the library, where they are called. My toolkit would not even precompile anymore

Reply:
Push the commit and copy the error. It's probably a syntax mistake.

Reply:
I pushed the commit 
```
julia> include("examples/ex010_advection.jl")
[ Info: Precompiling LFAToolkit [3f92b583-c0aa-4596-8bdf-f02f6c0a52df]
ERROR: LoadError: syntax: unexpected ")"
Stacktrace:
  [1] top-level scope
    @ ~/Projects/LFAToolkit.jl/src/Basis/Constructors.jl:139
  [2] include(mod::Module, _path::String)
    @ Base ./Base.jl:418
  [3] include(x::String)
    @ LFAToolkit ~/Projects/LFAToolkit.jl/src/LFAToolkit.jl:5
  [4] top-level scope
    @ ~/Projects/LFAToolkit.jl/src/LFAToolkit.jl:67
  [5] include
    @ ./Base.jl:418 [inlined]
  [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1318
  [7] top-level scope
    @ none:1
  [8] eval
    @ ./boot.jl:373 [inlined]
  [9] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [10] top-level scope
    @ none:1
in expression starting at /home/adebanks/Projects/LFAToolkit.jl/src/Basis/Constructors.jl:139
in expression starting at /home/adebanks/Projects/LFAToolkit.jl/src/LFAToolkit.jl:5
ERROR: LoadError: Failed to precompile LFAToolkit [3f92b583-c0aa-4596-8bdf-f02f6c0a52df] to /home/adebanks/.julia/compiled/v1.7/LFAToolkit/jl_6kcYSv.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1466
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1410
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1120
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [8] top-level scope
   @ REPL[1]:1
in expression starting at /home/adebanks/Projects/LFAToolkit.jl/examples/ex010_advection.jl:11
```